index.ts

```diff
@@ -1,33 +1,43 @@
-require('dotenv').config()
-import { Client, Intents } from 'discord.js'
-import dbConnect from './components/database'
-import Guild from './models/Guild'
-import HelpCommand from './commands/help'
+import {config} from "dotenv";
+import { Client, Intents } from 'discord.js';
+import dbConnect from './components/database';
+import Guild from './models/Guild';
+import HelpCommand from './commands/help';
 import PromoteNewJob from './components/promoteNewJob'
 import GetDefaultChannel from './components/getDefaultChannel'
 import WelcomeMessage from './components/formatting/welcome'
-
 import express from 'express'
+import bodyParser from 'body-parser'
+
+
+config({path: process.env.NODE_ENV === "dev" ? "./.env.dev":"./.env.prod"});
```

Comment:
might be better to use https://github.com/kerimdzhanov/dotenv-flow instead

Reply:
you also might want to import env config as the very first thing. many components might be dependent on it